src/plugins/data_source_management/server/routes/data_connections_router.ts

```diff
@@ -222,7 +222,9 @@ export function registerDataConnectionsRoute(router: IRouter, dataSourceEnabled:
         let dataConnectionsresponse;
         if (dataSourceEnabled && dataSourceMDSId) {
           const client = await context.dataSource.opensearch.legacy.getClient(dataSourceMDSId);
-          dataConnectionsresponse = await client.callAPI('ppl.getDataConnections');
+          dataConnectionsresponse = await client.callAPI('ppl.getDataConnections', {
+            requestTimeout: 2000, // Enforce timeout to avoid hanging requests
```

Comment:
Couldn't we add this as a config

Reply:
this is already a config, but for any new client. Here's a special handling to avoid long hanging calls